tests/Unit/NumericalAlgorithms/DiscontinuousGalerkin/Actions/Test_InitializeDomain.cpp

```diff
@@ -0,0 +1,296 @@
+#include "Domain/Creators/Brick.hpp"
+#include "Domain/Creators/Interval.hpp"
+#include "Domain/Creators/Rectangle.hpp"
+#include "Domain/Direction.hpp"
+#include "Domain/Domain.hpp"
+#include "Domain/Element.hpp"
+#include "Domain/ElementId.hpp"
+#include "Domain/ElementIndex.hpp"
+#include "Domain/ElementMap.hpp"
+#include "Domain/Mesh.hpp"
+#include "Domain/SegmentId.hpp"
+#include "Domain/Tags.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Actions/InitializeDomain.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Parallel/AddOptionsToDataBox.hpp"
+#include "Utilities/TMPL.hpp"
+#include "tests/Unit/ActionTesting.hpp"
+
+namespace {
+
+template <size_t Dim, typename Metavariables>
+struct ElementArray {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = ElementIndex<Dim>;
+  using const_global_cache_tag_list = tmpl::list<>;
+  using add_options_to_databox = Parallel::AddNoOptionsToDataBox;
+  using phase_dependent_action_list = tmpl::list<
+      Parallel::PhaseActions<
+          typename Metavariables::Phase, Metavariables::Phase::Initialization,
+          tmpl::list<ActionTesting::InitializeDataBox<
+              tmpl::list<::Tags::Domain<Dim, Frame::Inertial>,
+                         ::Tags::InitialExtents<Dim>>>>>,
+
+      Parallel::PhaseActions<
+          typename Metavariables::Phase, Metavariables::Phase::Testing,
+          tmpl::list<dg::Actions::InitializeDomain<
+              Dim, Initialization::MergePolicy::IgnoreIncomparable>>>>;
+};
+
+template <size_t Dim>
+struct Metavariables {
+  using component_list = tmpl::list<ElementArray<Dim, Metavariables>>;
+  using const_global_cache_tag_list = tmpl::list<>;
+  enum class Phase { Initialization, Testing, Exit };
+};
+
+template <size_t Dim>
+void check_compute_items(
+    const ActionTesting::MockRuntimeSystem<Metavariables<Dim>>& runner,
+    const ElementId<Dim>& element_id) {
+  // The compute items themselves are tested elsewhere, so just check if they
+  // were indeed added by the initializer
+  const auto tag_is_retrievable = [&runner,
+                                   &element_id](auto tag_v) -> decltype(auto) {
+    using tag = std::decay_t<decltype(tag_v)>;
+    return ActionTesting::tag_is_retrievable<
+        ElementArray<Dim, Metavariables<Dim>>, tag>(runner, element_id);
+  };
+  CHECK(tag_is_retrievable(::Tags::Coordinates<Dim, Frame::Logical>{}));
+  CHECK(tag_is_retrievable(::Tags::Coordinates<Dim, Frame::Inertial>{}));
+  CHECK(tag_is_retrievable(
+      ::Tags::InverseJacobian<::Tags::ElementMap<Dim>,
+                              ::Tags::Coordinates<Dim, Frame::Logical>>{}));
+  CHECK(tag_is_retrievable(::Tags::MinimumGridSpacing<Dim, Frame::Inertial>{}));
+}
+
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Domain.Actions.InitializeDomain",
+                  "[Unit][Domain][Actions]") {
+  {
+    INFO("1D");
+    // Reference element:
+    // [ |X| | ] -xi->
```

Comment:
I don't really get this ascii diagram, what is it trying to convey?

Reply:
It shows the domain that is set up in the following lines. We are testing the element with index 1 at refinement level 2. How is this unclear, or do you have a suggestion how to make it clearer? Similar diagrams are used in other tests.

Reply:
Alright, it makes more sense looking at the 2d one. Is the minus sign in front of `xi` intentional? Looking at the element ID (2nd out of 4 elements) and the map (-0.5, 1.5) -> (2, 4) doesn't suggest the `xi` axis is being flipped to me; maybe I'm missing something.

Reply:
It's not supposed to be a minus sign but part of the arrow 😄 But I see how this is misleading, will change it